src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -275,17 +289,77 @@ def _get_prioritized_spans(
             span_metadata_size = spans_to_send_sizes[index]
             span_size = get_event_base64_size(span)
 
-            if current_size + span_size - span_metadata_size < request_max_size:
+            if current_size + span_size - span_metadata_size < buffered_max_size:
                 spans_to_send_dict[index] = span
                 current_size += span_size - span_metadata_size
             else:
                 # This is an optimization step. If the spans are too big, don't try to send them.
                 too_big_spans += 1
                 if too_big_spans >= too_big_spans_threshold:
                     break
+
+            # If we dropped spans we need to update the enrichment spans dropped spans reasons
+            if len(spans_to_send_dict) != len(msgs):
+                with lumigo_safe_execute(
+                    "create_request_body: smart span selection: updating enrichment span"
+                ):
+                    spans_to_send_dict = _update_enrichment_span_about_prioritized_spans(
+                        spans_to_send_dict, msgs, current_size, buffered_max_size
+                    )
+
     return list(spans_to_send_dict.values())
 
 
+def _update_enrichment_span_about_prioritized_spans(
```

Comment:
I think you should try to rewrite this method the goals should be
1. avoid iterating the spans many times
2. try to not update dictionaries inside span
3. make sure we cant cross the limit size

my suggestion is to not add the enrichment span to the list but keep space for it and just add it once in the end.

Reply:
Done